### DISCUSSION TITLE
Exclude errors by pattern

### DISCUSSION THREAD

**Discussion by andrii-pez** [JetBrains/Qodana#94] [Q&A]
Hi! Thanks for the useful product.
Is it possible to hide errors by some text pattern, for instance using regular expression(or similar)?

**Comment by brichbash** [comment]
Hi,

Do you mean excluding errors based on their description? There is no way to filter problems using regexp yet.

At this moment, you can only exclude inspections for specified files or directories using `qodana.yaml` (can be also done via Qodana UI, then you just need to put changed `qodana.yaml` to your project root). Here are [some docs](https://www.jetbrains.com/help/qodana/qodana-yaml.html#Set+up+a+profile) on customizing your inspection profile.

**Comment by andrii-pez** [comment]
I already use `qodana.yaml` and list of ID's which should be ignored for some files, but it would be cool to ignore not all errors with some ID, but particular. For instance I have the same error `Parameter '$this->xxxx' type is not compatible with declaration` appearing few times in code. Off course I can add `PhpParamsInspection` ID to `exclude` section in profile, but I don't want to ignore all such errors, just want to hide error exactly with text `Parameter '$this->xxxx' type is not compatible with declaration`

**Comment by tiulpin** [comment]
You also can suppress the inspection results in your codebase, but this needs to be done in the IDE – https://www.jetbrains.com/help/phpstorm/disabling-and-enabling-inspections.html#suppress-inspections

**Comment by esafirm** [comment]
Hi, just want to check if there's plan to add this functionality in the future. Thanks

My use case is to exclude files that related to annotation processor.

**Comment by fernanfs** [comment]
Looking at this discussion and the current documentation, I'm a bit surprised that this seems to be a missing feature.

In my case, I have a go codebase that has a lot of unit tests alongside the actual go code. In these there are a number of findings like embedded credentials or similar (testing database connection in that case, that is started prior to the test cases).

Unfortunately, I can't see a simple way to exclude these files from the checking. Or am I missing something? Yes, I could add all test files individually, but that doesn't seem efficient at all.
